byway/src/main/webapp/track-interests.js

```diff
@@ -12,13 +12,43 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+let interestsChosen = new Set();
+let tripId;
+
 if(document.readyState === 'loading') {
-  document.addEventListener('DOMContentLoaded', loadButtons);
+  document.addEventListener('DOMContentLoaded', loadContent);
 } else {
+  loadContent();
+}
+
+/**
+ * Loads the content of the page by loading buttons
+ * and the form tracking buttons selected.
+ */
+function loadContent() {
+  loadId();
   loadButtons();
+  loadForm();
 }
 
-let interestsChosen = new Set();
+/* Parse url to retrieve the trip id. */
+function loadId() {
```

Comment:
Try to be more descriptive: `getTripIdFromUrl`.

Reply:
Updated